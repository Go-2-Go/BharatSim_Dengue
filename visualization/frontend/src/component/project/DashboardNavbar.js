import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import React from 'react';
import Box from '@material-ui/core/Box';
import { makeStyles } from '@material-ui/core/styles';
import dashboardIcon from '../../assets/images/dashboard-icon.svg';

const tabStyles = makeStyles((theme) => ({
  wrapper: {
    display: 'flex',
    flexDirection: 'row',
    '& > *:first-child': {
      marginBottom: '0 !important',
    },
    justifyContent: 'left',
  },
  labelIcon: {
    minHeight: '40px',
    height: '40px',
  },
  selected: {
    background: 'rgba(255, 255, 255,0.2)',
  },
  root: {
    padding: theme.spacing(3, 8, 3, 8),
    height: theme.spacing(10),
    borderRadius: theme.spacing(1),
  },
}));

const tabsStyles = makeStyles(() => ({
  indicator: {
    background: 'transparent',
  },
}));

export default function DashboardNavbar({ controllers, handleChange, value }) {
  const tabClasses = tabStyles();
  const tabsClasses = tabsStyles();
  return (
    <Box px={2}>
      <Tabs
        orientation="vertical"
        variant="fullWidth"
        value={value}
        onChange={handleChange}
        classes={tabsClasses}
      >
        {controllers.map((controller) => (
          <Tab
            icon={
              <Box pr={3} display="flex" alignItems="center">
                <img src={dashboardIcon} alt="dashboard-logo" />
              </Box>
            }
            label={controller}
            key={`controller-${controller}`}
            classes={tabClasses}
            data-testid={`tab-${controller}`}
          />
        ))}
      </Tabs>
    </Box>
  );
}
