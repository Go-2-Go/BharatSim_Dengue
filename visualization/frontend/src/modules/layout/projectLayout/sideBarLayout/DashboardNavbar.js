import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import React from 'react';
import Box from '@material-ui/core/Box';
import { makeStyles } from '@material-ui/core/styles';
import dashboardIcon from '../../../../assets/images/dashboard-icon.svg';

const tabStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(3, 7),
    height: theme.spacing(10),
    borderRadius: theme.spacing(1),
    minHeight: 'unset',
    minWidth: 'unset',
    boxSizing: 'border-box',
    ...theme.typography.subtitle2,
    [theme.breakpoints.up('sm')]: {
      minWidth: 'unset',
    },
  },
  wrapper: {
    display: 'flex',
    flexDirection: 'row',
    '& > *:first-child': {
      marginBottom: '0 !important',
    },
    justifyContent: 'left',
  },
  labelIcon: {
    minHeight: '40px',
    height: '40px',
  },
  selected: {
    background: theme.colors.primaryColorScale['500'],
  },
  textColorInherit: {
    border: 'none',
    color: theme.colors.textLight.primary,
    textTransform: 'capitalize',
    opacity: 'unset',
    '&$selected': {
      border: 'none',
      opacity: 'unset',
      color: theme.colors.textLight.primary,
    },
  },
}));

const tabsStyles = makeStyles(() => ({
  indicator: {
    background: 'transparent',
  },
}));

/* TODO Add onchange for selecting dashboard */

export default function DashboardNavbar({ navItems, value }) {
  const tabClasses = tabStyles();
  const tabsClasses = tabsStyles();
  return (
    <Box px={2}>
      <Tabs orientation="vertical" variant="fullWidth" value={value} classes={tabsClasses}>
        {navItems.map((controller) => (
          <Tab
            icon={(
              <Box pr={3} display="flex" alignItems="center">
                <img src={dashboardIcon} alt="dashboard-logo" />
              </Box>
            )}
            label={controller}
            key={`controller-${controller}`}
            classes={tabClasses}
            data-testid={`tab-${controller}`}
          />
        ))}
      </Tabs>
    </Box>
  );
}
