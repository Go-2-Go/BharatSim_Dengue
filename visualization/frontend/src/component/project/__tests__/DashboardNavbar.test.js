import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import DashboardNavbar from '../DashboardNavbar';

describe('Dashboard controller', function () {
  it('should match snapshot', function () {
    const handleChange = jest.fn();
    const { container } = render(
      <DashboardNavbar
        value={0}
        handleChange={handleChange}
        controllers={['dashboard1', 'dashboard2', 'dashboard3']}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('should call handle change on click of tab', function () {
    const handleChange = jest.fn();
    const { getByTestId } = render(
      <DashboardNavbar
        value={0}
        handleChange={handleChange}
        controllers={['dashboard1', 'dashboard2', 'dashboard3']}
      />,
    );
    const dashboard1Tab = getByTestId('tab-dashboard1');

    fireEvent.click(dashboard1Tab);
    expect(handleChange).toHaveBeenCalled();
  });
});
