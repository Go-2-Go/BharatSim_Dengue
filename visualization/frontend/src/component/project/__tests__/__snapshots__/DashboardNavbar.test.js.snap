// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard controller should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <div
      class="MuiTabs-root MuiTabs-vertical"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          class="MuiTabs-flexContainer MuiTabs-flexContainerVertical"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root makeStyles-root-4 MuiTab-textColorInherit Mui-selected makeStyles-selected-3 MuiTab-labelIcon makeStyles-labelIcon-2 MuiTab-fullWidth"
            data-testid="tab-dashboard1"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper makeStyles-wrapper-1"
            >
              <div
                class="MuiBox-root MuiBox-root-7"
              >
                <img
                  alt="dashboard-logo"
                  src="dashboard-icon.svg"
                />
              </div>
              dashboard1
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root makeStyles-root-4 MuiTab-textColorInherit MuiTab-labelIcon makeStyles-labelIcon-2 MuiTab-fullWidth"
            data-testid="tab-dashboard2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper makeStyles-wrapper-1"
            >
              <div
                class="MuiBox-root MuiBox-root-12"
              >
                <img
                  alt="dashboard-logo"
                  src="dashboard-icon.svg"
                />
              </div>
              dashboard2
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root makeStyles-root-4 MuiTab-textColorInherit MuiTab-labelIcon makeStyles-labelIcon-2 MuiTab-fullWidth"
            data-testid="tab-dashboard3"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper makeStyles-wrapper-1"
            >
              <div
                class="MuiBox-root MuiBox-root-13"
              >
                <img
                  alt="dashboard-logo"
                  src="dashboard-icon.svg"
                />
              </div>
              dashboard3
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-8 PrivateTabIndicator-colorSecondary-10 MuiTabs-indicator makeStyles-indicator-5 PrivateTabIndicator-vertical-11"
          style="top: 0px; height: 0px;"
        />
      </div>
    </div>
  </div>
</div>
`;
